smsclient_core/models/sms_gateway.py

```diff
@@ -0,0 +1,226 @@
+
+import logging
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+PRIORITY_LIST = [
+    ('0', '0'),
+    ('1', '1'),
+    ('2', '2'),
+    ('3', '3')
+]
+
+CLASSES_LIST = [
+    ('0', 'Flash'),
+    ('1', 'Phone display'),
+    ('2', 'SIM'),
+    ('3', 'Toolkit')
+]
+
+
+class SMSClient(models.Model):
+    _name = 'sms.gateway'
+    _description = 'SMS Client'
+
+    @api.model
+    def get_method(self):
+        return []
+
+    name = fields.Char(string='Gateway Name', required=True)
+    from_provider = fields.Char(string="From")
+    method = fields.Selection(
+        string='API Method',
+        selection='get_method')
+    url = fields.Char(
+        string='Gateway URL', help='Base url for message')
+    state = fields.Selection(selection=[
+        ('new', 'Not Verified'),
+        ('waiting', 'Waiting for Verification'),
+        ('confirm', 'Verified'),
+        ], string='Gateway Status', index=True, readonly=True, default='new')
+    user_ids = fields.Many2many(
+        comodel_name='res.users',
+        string='Users Allowed')
+    code = fields.Char('Verification Code')
+    body = fields.Text(
+        string='Message',
+        help="The message text that will be send along with the"
+             " email which is send through this server.")
+    validity = fields.Integer(
+        default=10,
+        help="The maximum time - in minute(s) - before the message "
+             "is dropped.")
+    classes = fields.Selection(
+        selection=CLASSES_LIST, string='Class',
+        default='1',
+        help='The SMS class: flash(0),phone display(1),SIM(2),toolkit(3)')
+    deferred = fields.Integer(
+        default=0,
+        help='The time -in minute(s)- to wait before sending the message.')
+    deferred_visible = fields.Boolean(default=False)
+    priority = fields.Selection(
+        selection=PRIORITY_LIST, string='Priority', default='3',
+        help='The priority of the message')
+    coding = fields.Selection(selection=[
+        ('1', '7 bit'),
+        ('2', 'Unicode')
+        ], string='Coding',
+        help='The SMS coding: 1 for 7 bit (160 chracters max'
+             'lenght) or 2 for unicode (70 characters max'
+             'lenght)',
+        default='1'
+    )
+    tag = fields.Char('Tag', help='an optional tag')
+    nostop = fields.Boolean(
+        default=True,
+        help='Do not display STOP clause in the message, this requires that '
+             'this is not an advertising message.')
+    char_limit = fields.Boolean('Character Limit', default=True)
+    default_gateway = fields.Boolean(default=False)
+    company_id = fields.Many2one(comodel_name='res.company')
+
+    @api.multi
+    def _check_permissions(self):
+        self.ensure_one()
+        if self.env.uid not in self.sudo().user_ids.ids:
+            return False
+        return True
+
+    @api.model
+    def _run_send_sms(self, domain=None):
```

Comment:
This is functionality in our linter. You should never use a mutable object as a default parameter.

```
>>> class A:
...   def do_something(self, arg=[]):
...     arg.append('Test')
...     return arg
... 
>>> a = A()
>>> a.do_something()
['Test']
>>> a.do_something()
['Test', 'Test']
```